### PR TITLE
Map with data

### LINKED ISSUE #54: Map hovercard
For this ticket please finalize the map hovercard. It should be pulling in the state data dynamically via the API, and match the list view of the state.

**Hovercard:**
Hovercard

### LINKED ISSUE #107: Clicking a state on map goes to state page

### LINKED ISSUE #110: Pull real data into map (state color matches grade)

### LINKED ISSUE #112: Pull real data into state scorecard

### LINKED ISSUE #125: Pull overall state score descriptions into state list on homepage
Currently the placeholder text _"This state does not prioritize..."_ is being pulled in for each state in the state list on the homepage. This should be replaced with the same blurb that goes on the state page under the overall score. In the scorecard that is referenced as `overallScoreLabels[stateData.grade.grade]`.

Screen Shot 2021-03-02 at 9 02 31 PM

### PR DESCRIPTION
**Pull request by landonreed**
This PR:
- connects the map with data 
- adds in the hovercard
- Redirects to state page on clicking state in map view
- adds overall score label to states list/hovercard

Closes #54, #110, #107, #125.

Note: this is based on #132 (so that we can make use of `toSlug`).

FYI, @egiurleo, this PR breaks out `StateCard` into its own component (just mentioning since I think you may be working on #126 at some point).

image

### CODE REVIEW

**Review by sarahgreen — APPROVED** [review]
Awesome progress!! I'm fine with merging this in and filing separate tickets for polishing things up, but I'll leave it up to you two @landonreed @egiurleo.

**Review by egiurleo — APPROVED** [review]
This is awesome! Feel free to merge once the `toSlug` change is merged as well. Let me know if you need help resolving conflicts or anything :)

**Pull request merged**
